crates/core/tedge_mapper/src/c8y/tests.rs

```diff
@@ -456,24 +455,22 @@ async fn c8y_mapper_alarm_with_custom_fragment_mapping_to_c8y_json() {
 
     broker
         .publish_with_opts(
-            "tedge/alarms/major/temperature_alarm",
+            "tedge/alarms/major/custom_temperature_alarm",
             r#"{ "text": "Temperature high","time":"2023-01-25T18:41:14.776170774Z","customFragment": {"nested":{"value": "extra info"}} }"#,
             mqtt_channel::QoS::AtLeastOnce,
             true,
         )
         .await
         .unwrap();
 
-    let expected_msg = json!({"severity":"MAJOR","type":"temperature_alarm","time":"2023-01-25T18:41:14.776170774Z","text":"Temperature high","customFragment":{"nested":{"value":"extra info"}}});
-
-    while let Ok(Some(msg)) = messages.next().with_timeout(TEST_TIMEOUT_MS).await {
-        assert_json_include!(actual:serde_json::from_str::<serde_json::Value>(&msg).unwrap(), expected:expected_msg);
-    }
+    let expected_msg = r#"{"severity":"MAJOR","type":"custom_temperature_alarm","time":"2023-01-25T18:41:14.776170774Z","text":"Temperature high","customFragment":{"nested":{"value":"extra info"}}}"#;
```

Comment:
I'm assuming that you converted this to raw string to use the `assert_received_all_expected` function that expects string. You could have used the `to_string()` function on the earlier `json::Value` as well, right? So that you get a string without affecting the code readability. It doesn't matter in this specific case as the JSON was not formatted properly in the first place.

Reply:
Yes, thats right. I felt converting from `json` to string is not necessary if raw string is used directly.

Reply:
Using `json!`  is just for better readability, especially with multi-level JSON structs. More of a good to have feature than a must.